### PR TITLE
Render perf optimisations

### PR DESCRIPTION
**Pull request by metalgearsloth**
Not sure how much value it is at this stage.

### CODE REVIEW

File: OpenDreamClient/Rendering/DMISpriteComponent.cs [hunk 1]
```diff
@@ -45,7 +45,8 @@ sealed class DMISpriteComponent : SharedDMISpriteComponent, ILookupWorldBox2Comp
                 if (!_entityManager.TryGetComponent<TransformComponent>(Owner, out var transform))
                     return false;
 
-                EntityUid mapEntity = IoCManager.Resolve<IMapManager>().GetMapEntityId(transform.MapID);
+                IoCManager.Resolve(ref mapManager);
```
Comment (metalgearsloth) [inline OpenDreamClient/Rendering/DMISpriteComponent.cs:48]: IoC dependency collections are backed by ThreadLocals which add up if you have to resolve it heaps, in ss14 we just add nullable arguments as an easy way around it if the class isn't ECS.

File: OpenDreamClient/Rendering/DreamViewOverlay.cs [hunk 2]
```diff
@@ -1,57 +1,58 @@
...
-        private void DrawMap(DrawingHandleWorld handle, EntityUid eye) {
-
+        private void DrawMap(OverlayDrawArgs args, EntityUid eye) {
 
+            _transformSystem ??= _entitySystem.GetEntitySystem<SharedTransformSystem>();
             _lookupSystem ??= _entitySystem.GetEntitySystem<EntityLookupSystem>();
+            var spriteQuery = _entityManager.GetEntityQuery<DMISpriteComponent>();
+            var xformQuery = _entityManager.GetEntityQuery<TransformComponent>();
 
             var entities = _lookupSystem.GetEntitiesInRange(eye, 15);
```
Comment (metalgearsloth) [inline OpenDreamClient/Rendering/DreamViewOverlay.cs:35]: If the eye is attached to the player entity then you could just use the OverlayDrawArgs Mapid + WorldAABB instead?

Comment (wixoaGit) [reply OpenDreamClient/Rendering/DreamViewOverlay.cs:35]: That does appear to be a better way to do it, wasn't aware of it

**Review by wixoaGit — APPROVED** [review]

File: OpenDreamClient/Rendering/DreamViewOverlay.cs (hunk 2 above)
Comment (metalgearsloth) [reply OpenDreamClient/Rendering/DreamViewOverlay.cs:35]: Either MapID or WorldAABB were just added in the most recent engine update I think so probably not available sooner.

**Pull request merged**